### PR TITLE
bulk filter support

### PR DESCRIPTION
**Pull request by tomknig**
This PR adds some options to the `Filter` component for allowing the toggling of the visibility of all filters at once and some extensions to allow necessary style changes.

### CODE REVIEW

**Review by ArneZsng — APPROVED** [review]

File: packages/ra-core/src/controller/ListController.js [hunk 1]
```diff
@@ -69,7 +69,18 @@ import removeKey from '../util/removeKey';
  *     );
  */
 export class ListController extends Component {
-    state = {};
+    constructor(props) {
+        super(props);
+
+        const enabledSources = {};
+        props.initiallyEnabledSources.forEach(
```
Comment (JakeDluhy) [inline packages/ra-core/src/controller/ListController.js]: Good place for reduce:
```js
const enabledSources = props.initiallyEnabledSources.reduce((sources, src) => ({ ...sources, [src]: true }), {})
```
(or some multi line version). You could also do
```js
const enabledSources = _.fromPairs(props.initiallyEnabledSources.map((src) => [src, true]))
```

File: packages/ra-core/src/controller/ListController.js [hunk 2]
```diff
@@ -194,8 +207,18 @@ export class ListController extends Component {
         }
     };
 
+    hideActiveFilters = () => {
+        this.setState({ activeFilters: {} });
+    };
+
+    showInactiveFilters = () => {
```
Comment (JakeDluhy) [inline packages/ra-core/src/controller/ListController.js:214]: Should this be `showActiveFilters`?

**Review by JakeDluhy — CHANGES_REQUESTED** [review]
This is tough for me to parse without having a lot of the context of the library, but in general it looks good!

File: packages/ra-core/src/controller/ListController.js (hunk 1 above)
Comment (ArneZsng) [reply packages/ra-core/src/controller/ListController.js]: I like the reduce here and added it. @tomknig just update it in a follow up PR if you prefer something different.

File: packages/ra-core/src/controller/ListController.js (hunk 2 above)
Comment (ArneZsng) [reply packages/ra-core/src/controller/ListController.js:214]: @tomknig Maybe you can clarify later what happens here. If we need to change the wording, let's do it in a follow up PR.

**Pull request merged**